main/integration-tests/__test__/api-tests/common/advanced-tests/rstudio/launch-rstudio-workspace.test.js

```diff
@@ -56,28 +56,34 @@ describe('Launch and terminate RStudio instance', () => {
           return workspace.id;
         }),
       );
-      const workspacesInAvailableState = response.filter(workspace => {
-        return workspace.envTypeId === rstudioEnvTypeId && workspace.status === 'COMPLETED';
-      });
-      console.log('COMPLETED (Available) workspaces', workspacesInAvailableState);
-      for (let i = 0; i < workspacesInAvailableState.length; i += 1) {
-        console.log(`Terminating ${workspacesInAvailableState[i].id}`);
-        // eslint-disable-next-line no-await-in-loop
-        await adminSession.resources.workspaceServiceCatalogs
-          .workspaceServiceCatalog(workspacesInAvailableState[i].id)
-          .delete();
-      }
     }
     if (workspaces.length > 0) {
       throw new Error('All RStudio workspaces should be terminated or failed');
     }
   }
 
+  async function terminateAllRStudioWorkspacesInCompletedState() {
+    const response = await adminSession.resources.workspaceServiceCatalogs.get();
+    const rstudioEnvTypeId = setup.defaults.envTypes.rstudio.envTypeId;
+    const workspacesInAvailableState = response.filter(workspace => {
+      return workspace.envTypeId === rstudioEnvTypeId && workspace.status === 'COMPLETED';
+    });
+    console.log('COMPLETED (Available) workspaces', workspacesInAvailableState);
+    for (let i = 0; i < workspacesInAvailableState.length; i += 1) {
+      console.log(`Terminating ${workspacesInAvailableState[i].id}`);
+      // eslint-disable-next-line no-await-in-loop
+      await adminSession.resources.workspaceServiceCatalogs
+        .workspaceServiceCatalog(workspacesInAvailableState[i].id)
+        .delete();
+    }
+  }
+
   // eslint-disable-next-line jest/expect-expect
   it('should launch a RStudio instance', async () => {
     if (setup.defaults.envTypes.rstudio.envTypeId === 'N/A') {
       return;
     }
+    await terminateAllRStudioWorkspacesInCompletedState();
```

Comment:
In your previous revision you were calling it towards the end of `checkAllRstudioWorkspaceIsTerminated`. So should this be called after `checkAllRstudioWorkspaceIsTerminated`?

Reply:
I think it's better to call `terminateAllRStudioWorkspacesInCompletedState` before `checkAllRstudioWorkspaceIsTerminated` because if there are workspaces that should be terminated, we should terminate them before polling whether all workspaces are terminated.

Reply:
Discussed offline and decided on moving `terminateAllRStudioWorkspacesInCompletedState` to beforeEach